### PR TITLE
Add CI and github config

### PR DESCRIPTION
**Pull request by nightkr**
Mostly copied from operator-rs. Would be nice to have this managed by operator-templating, but that depends on https://github.com/stackabletech/operator-templating/issues/80.

See #11.

### CODE REVIEW

**Comment by nightkr** [comment]
The remaining pre-commit failures are from issues in the common files, so we should probably address those upstream instead...

Doing a followup PR to address the cargo deny failures.

**Comment by nightkr** [comment]
The remaining deny failures (from `users`) are addressed in #13.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+          submodules: recursive
+      - uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: ${{ env.RUST_TOOLCHAIN_VERSION }}
+          components: clippy
+      - uses: Swatinem/rust-cache@82a92a6e8fbeee089604da2575dc567ae9ddeaab # v2.7.5
+        with:
+          key: clippy
+      - name: Run clippy action to produce annotations
+        # NOTE (@Techassi): This action might get a new release sonn, because it
```
Comment (razvan) [inline .github/workflows/build.yml]:
```suggestion
        # NOTE (@Techassi): This action might get a new release soon, because it
```

**Review by razvan — DISMISSED** [review]
lgtm

**Review by razvan — APPROVED** [review]

**Pull request merged**